modules/imaging_browser/php/viewsession.class.inc

```diff
@@ -175,11 +175,11 @@ class ViewSession extends \NDB_Form
             "SELECT files.FileID, files.ScannerID FROM files
              LEFT JOIN files_qcstatus as sel
              on (files.FileID=sel.FileID)
-            WHERE SessionID=:SID AND (AcquisitionProtocolID IS NULL
-            OR AcquisitionProtocolID not in (1, 2, 3, 52))
+            WHERE SessionID=:SID AND (MriScanTypeID IS NULL
+            OR MriScanTypeID not in (1, 2, 3, 52))
             $extra_where_string
             ORDER BY files.OutputType,
-                     files.AcquisitionProtocolID,
+                     files.MriScanTypeID,
                      files.File",
             $args
         );
```

Comment:
When testing view session page, I get the following error when accessing `imaging_browser/viewSession/?sessionID=2144&outputType=native&backURL=/imaging_browser/`:
```
[Tue Nov 12 10:32:35.067078 2024] [php:notice] [pid 57283:tid 57283] [client 192.168.122.1:51354] [ERROR] SQLSTATE[HY000]: General error: 2014 Cannot execute queries while there are pending result sets. Consider unsetting the previous PDOStatement or calling PDOStatement::closeCursor()#0 /var/www/loris/src/Database/Query.php(42): PDOStatement->execute()\n#1 /var/www/loris/src/Database/Query.php(33): LORIS\\Database\\Query->getIterator()\n#2 /var/www/loris/php/libraries/Database.class.inc(840): LORIS\\Database\\Query->getFirstRow()\n#3 /var/www/loris/php/libraries/Database.class.inc(979): Database->pselectRow()\n#4 /var/www/loris/php/libraries/Database.class.inc(1008): Database->pselectOne()\n#5 /var/www/loris/modules/imaging_browser/php/viewsession.class.inc(397): Database->pselectOneInt()\n#6 /var/www/loris/modules/imaging_browser/php/viewsession.class.inc(310): LORIS\\imaging_browser\\ViewSession->_getManualCaveatViolationsResolvedID()\n#7 /var/www/loris/modules/imaging_browser/php/viewsession.class.inc(116): LORIS\\imaging_browser\\ViewSession->_setFilesData()\n#8 /var/www/loris/php/libraries/NDB_Form.class.inc(133): LORIS\\imaging_browser\\ViewSession->setup()\n#9 /var/www/loris/src/Middleware/UserPageDecorationMiddleware.php(248): NDB_Form->handle()\n#10 /var/www/loris/src/Middleware/PageDecorationMiddleware.php(59): LORIS\\Middleware\\UserPageDecorationMiddleware->process()\n#11 /var/www/loris/php/libraries/NDB_Page.class.inc(726): LORIS\\Middleware\\PageDecorationMiddleware->process()\n#12 /var/www/loris/php/libraries/Module.class.inc(322): NDB_Page->process()\n#13 /var/www/loris/src/Middleware/ResponseGenerator.php(51): Module->handle()\n#14 /var/www/loris/src/Middleware/AuthMiddleware.php(64): LORIS\\Middleware\\ResponseGenerator->process()\n#15 /var/www/loris/src/Router/ModuleRouter.php(75): LORIS\\Middleware\\AuthMiddleware->process()\n#16 /var/www/loris/src/Middleware/ExceptionHandlingMiddleware.php(55): LORIS\\Router\\ModuleRouter->handle()\n#17 /var/www/loris/src/Router/BaseRouter.php(138): LORIS\\Middleware\\ExceptionHandlingMiddleware->process()\n#18 /var/www/loris/src/Middleware/ResponseGenerator.php(51): LORIS\\Router\\BaseRouter->handle()\n#19 /var/www/loris/src/Middleware/ContentLength.php(53): LORIS\\Middleware\\ResponseGenerator->process()\n#20 /var/www/loris/htdocs/index.php(74): LORIS\\Middleware\\ContentLength->process()\n#21 {main}, referer: https://cecile-dev-new.loris.ca/imaging_browser/?sequenceType=fMRI
```

Some viewSession pages are fine, others return this error.

Reply:
Hhhhm, I've been having this error occasionally since #9334 got merged. Honestly it is hard to debug, and may be present on main ? Blows but I'll look into it when I have time.

Reply:
let me check on main. I actually was about to do that but it was time of the LORIS meeting and then forgot to check...

Reply:
#9465